core/src/main/java/org/bf2/srs/fleetmanager/execution/impl/workers/ScheduleRegistryWorker.java

```diff
@@ -88,17 +117,59 @@ public void execute(Task aTask, WorkerContext ctl) throws RegistryStorageConflic
         ctl.delay(() -> tasks.submit(ProvisionRegistryTenantTask.builder().registryId(registry.getId()).build()));
     }
 
+    private RegistryData createRegistry(RegistryData registryData, AccountInfo accountInfo) throws EvalInstancesNotAllowedException, AccountManagementServiceException, TooManyEvalInstancesForUserException, TermsRequiredException, ResourceLimitReachedException, RegistryStorageConflictException {
```

Comment:
I'd prefer this runs as a completely separate task (probably before this one runs). An issue here is propagating and handling of errors. When an async task fails (throws an exception), the task is retried, so the code needs to be re-entrant. Then the `finallyExecute` handler is *always* called. You have to handle anything that can go well or wrong in this method. Since we cannot propagate an exception to the user, when the `EvalInstancesNotAllowedException` is thrown we need to probably set the status to `FAILED`, so user can retry. However this status does not contain additional information, so we should probably add a `reason` field to `org.bf2.srs.fleetmanager.rest.publicapi.beans.Registry` or maybe an embedded `Error` entity.

Reply:
I was actually preparing the separate task before you commented so we're on the same page here. This was just the first round of changes while trying to keep everything up & running.